packages/monaco/src/browser/monaco-command.ts

```diff
@@ -7,16 +7,82 @@
 
 import { injectable, inject } from "inversify";
 import { ProtocolToMonacoConverter } from "monaco-languageclient/lib";
-import {
-    CommandHandler, CommandContribution, CommandRegistry, SelectionService
-} from '@theia/core/lib/common';
+import { Command, CommandHandler, CommandContribution, CommandRegistry, SelectionService } from '@theia/core';
+import { Position, Location } from "@theia/languages/lib/common";
 import { CommonCommands } from '@theia/core/lib/browser';
 import { EditorManager, TextEditorSelection, SHOW_REFERENCES } from '@theia/editor/lib/browser';
-import { Position, Location } from "@theia/languages/lib/common"
 import { getCurrent, MonacoEditor } from './monaco-editor';
 import MenuRegistry = monaco.actions.MenuRegistry;
 import MenuId = monaco.actions.MenuId;
 
+export type MonacoCommand = Command & { delegate?: string };
+export namespace MonacoCommands {
+
+    export const UNDO = 'undo';
+    export const REDO = 'redo';
+    export const COMMON_KEYBOARD_ACTIONS = new Set([UNDO, REDO]);
+    export const COMMON_ACTIONS: {
+        [action: string]: string
+    } = {};
+    COMMON_ACTIONS[UNDO] = CommonCommands.UNDO.id;
+    COMMON_ACTIONS[REDO] = CommonCommands.REDO.id;
+    COMMON_ACTIONS['actions.find'] = CommonCommands.FIND.id;
+    COMMON_ACTIONS['editor.action.startFindReplaceAction'] = CommonCommands.REPLACE.id;
+
+    export const SELECTION_MENU = '3_selection';
+
+    export const SELECTION_MENU_SELECTION_GROUP = '1_selection_group';
+    export const SELECTION_MENU_COPY_MOVE_GROUP = '2_copy_move_group';
+    export const SELECTION_MENU_CURSOR_GROUP = '3_cursor_group';
+
+    export const SELECTION_SELECT_ALL = 'editor.action.select.all';
```

Comment:
should be `monaco.action. ...`etc.

Reply:
That is the eventual monaco command we have to execute.

Reply:
Does that mean we need another mapping, that maps the `theia-monaco` command ID to the actual `monaco` command ID?

Reply:
We have to remove prefix before the execution.

Reply:
I am sorry, I did not get this. What prefix?